### DISCUSSION TITLE
FileNotFoundError: Could not find module

### DISCUSSION THREAD

**Discussion by JMoreno97** [pyinstaller/pyinstaller#7501] [Help] [answered]
### System

- Windows 10 64 bits
- Python 3.9.13

Good afternoon, I have generated the executable of my application using the following command: 

`pyinstaller --onefile --clean --debug=all --icon record.ico --name Recorder main.py`

### Error

When I run it the application closes and debugging a little I have seen the following error:

```
import 'screen_recorder_sdk.screen_recorder' # <_frozen_importlib_external.SourcelessFileLoader object at 0x0000017BA14C9310>
# _bootlocale not found in PYZ
import _bootlocale # loaded from Zip C:\Users\IEETA-XR\AppData\Local\Temp\_MEI108402\base_library.zip\_bootlocale.pyc
Traceback (most recent call last):
  File "PyInstaller\loader\pyimod03_ctypes.py", line 53, in __init__
  File "ctypes\__init__.py", line 374, in __init__
FileNotFoundError: Could not find module 'C:\Users\IEETA-XR\AppData\Local\Temp\_MEI108402\screen_recorder_sdk\lib\ScreenRecorder.dll' (or one of its dependencies). Try using the full path with constructor syntax.
```

I've looked everywhere and they always talk about adding the library in question to the hidden imports but I can't get it to work. 
The program is very simple, it uses a library (screen_recorder_sdk) to record the screen when asked. Here is the code below:

### Code
```
import os
import socket
import threading

from screen_recorder_sdk import screen_recorder


class ScreenRecorder(threading.Thread):
    def __init__(self, path):
        threading.Thread.__init__(self)
        self.shouldStop = False
        self.videoPath = path

    def run(self):
        screen_recorder.start_video_recording(self.videoPath)
        while True:
            if self.shouldStop:
                break
        screen_recorder.stop_video_recording()
        screen_recorder.free_resources()


screen_recorder.enable_dev_log()
screen_recorder.set_log_file('recorder.log')

params = screen_recorder.RecorderParams()
screen_recorder.init_resources(params)


sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)


server_address = ('localhost', 10000)
print('starting up on {} port {}'.format(*server_address))
sock.bind(server_address)


sock.listen(1)

recorder = None

while True:

    print('waiting for a connection')
    connection, client_address = sock.accept()
    try:
        print('connection from', client_address)

        while True:
            data = connection.recv(1024)
            print('received {!r}'.format(data))
            if data:
                dataStr = data.decode("utf-8")
                if dataStr.lower().__contains__("start"):
                    name = dataStr[dataStr.index(":") + 1:]
                    if not (os.path.exists("./Proyecto GJ_Data/Recordings")):
                        os.mkdir("./Proyecto GJ_Data/Recordings")
                    path = "./Proyecto GJ_Data/Recordings/" + name + ".mp4"
                    recorder = ScreenRecorder(path)
                    recorder.start()
                elif dataStr.lower().__contains__("stop"):
                    if recorder:
                        recorder.shouldStop = True
                        recorder.join()
            else:
                print('no data from', client_address)
                break

    finally:
        connection.close()
```

Thank you

**Comment by rokm** [comment, marked as answer]
> ```
> import 'screen_recorder_sdk.screen_recorder' # <_frozen_importlib_external.SourcelessFileLoader object at 0x0000017BA14C9310>
> # _bootlocale not found in PYZ
> import _bootlocale # loaded from Zip C:\Users\IEETA-XR\AppData\Local\Temp\_MEI108402\base_library.zip\_bootlocale.pyc
> Traceback (most recent call last):
>   File "PyInstaller\loader\pyimod03_ctypes.py", line 53, in __init__
>   File "ctypes\__init__.py", line 374, in __init__
> FileNotFoundError: Could not find module 'C:\Users\IEETA-XR\AppData\Local\Temp\_MEI108402\screen_recorder_sdk\lib\ScreenRecorder.dll' (or one of its dependencies). Try using the full path with constructor syntax.
> ```

Does adding `--collect-binaries screen_recorder_sdk` to your PyInstaller command help?

**Reply by JMoreno97** [reply]
Sorry for not replying sooner but I couldn't test it until today. This solved the error for me. I understand that when I get a missing .dll error that will be the option I should use.

Thank you very much

**Reply by PavanRajapantula** [reply]
Hey, can you please help me with this error ?

File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\main.py", line 4, in <module>
    from Class_Sturm import SlP
  File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\Class_Sturm.py", line 14, in <module>
    import solver
  File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\solver.py", line 5, in <module>
    lib = ctypes.cdll.LoadLibrary(lib_path)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\ctypes\__init__.py", line 460, in LoadLibrary
    return self._dlltype(name)
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\ctypes\__init__.py", line 379, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: Could not find module 'C:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\librungekutta.dll' (or one of its dependencies). Try using the full path with constructor syntax.

This is for the project [https://github.com/ShayanBajelan/Sturm-Liouville-Problem-GUI/tree/main] , can you please help me?

**Reply by PavanRajapantula** [reply]
Hey, can you please help me with this error ?

File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\main.py", line 4, in
from Class_Sturm import SlP
File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\Class_Sturm.py", line 14, in
import solver
File "c:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\solver.py", line 5, in
lib = ctypes.cdll.LoadLibrary(lib_path)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Python312\Lib\ctypes_init_.py", line 460, in LoadLibrary
return self.dlltype(name)
^^^^^^^^^^^^^^^^^^^
File "C:\Python312\Lib\ctypes_init.py", line 379, in init
self._handle = _dlopen(self._name, mode)
^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: Could not find module 'C:\Users\pavan\OneDrive\Desktop\Sturm-Liouville-Problem-GUI-main\librungekutta.dll' (or one of its dependencies). Try using the full path with constructor syntax.

This is for the project '[https://github.com/ShayanBajelan/Sturm-Liouville-Problem-GUI/tree/main]', can you please help me?
